### ISSUE TITLE
TypeError Cannot assign string to property verbb\snipcart\models\ProductDetails::$price of type ?float

### ISSUE THREAD

**Issue by elvdm** [verbb/snipcart-craft-plugin#67]
### Describe the bug

When editing an entry with the Snipcart Product Details field added to it, if you forget to put a value in the price field then the following error is thrown:

```
TypeError
Cannot assign string to property verbb\snipcart\models\ProductDetails::$price of type ?float
1. in /var/www/html/vendor/yiisoft/yii2/base/Model.phpat line 1047
1038103910401041104210431044104510461047104810491050105110521053105410551056     * Sets the element at the specified offset.
     * This method is required by the SPL interface [[\ArrayAccess]].
     * It is implicitly called when you use something like `$model[$offset] = $value;`.
     * @param string $offset the offset to set element
     * @param mixed $value the element value
     */
    #[\ReturnTypeWillChange]
    public function offsetSet($offset, $value)
    {
        $this->$offset = $value;
    }
 
    /**
     * Sets the element value at the specified offset to null.
     * This method is required by the SPL interface [[\ArrayAccess]].
     * It is implicitly called when you use something like `unset($model[$offset])`.
     * @param string $offset the offset to unset element
     */
    #[\ReturnTypeWillChange]
2. in /var/www/html/vendor/verbb/snipcart/src/validators/ProductDetailsValidator.php at line 30– [yii\base\Model::offsetSet](https://www.yiiframework.com/doc-2.0/yii-base-model.html#offsetSet()-detail)('price', '')
24252627282930313233343536        // don't like this but...
        // we are using a money field so when we try to save the value
        // there is an issue with the column type in the database (DECIMAL)
        // and it doesn't like the value being passed to it
        // so we remove the comma from the value
        // there's definitely a better way to do this ¯\_(ツ)_/¯
        $value['price'] = str_replace(',', '', $value['price']);
 
        // SKU field validations 
        // test for empty SKU
        if ($value['sku'] === null || trim($value['sku']) === '') {
            $this->addError($model, $attribute, 'SKU cannot be blank');
        }
3. in /var/www/html/vendor/yiisoft/yii2/validators/Validator.php at line 260– verbb\snipcart\validators\ProductDetailsValidator::validateAttribute(craft\elements\Entry, 'field:productDetails')
```

Would expect instead of throwing an error, that the entry would not be saved and you should get a validation error for the field instead. It saves fine if you don't forget to fill out the price.

### Steps to reproduce

1. Add Snipcart Product Details field to an entry type
2. Edit an entry with the Snipcart Product Details field on it
3. Leave "price" blank and try to save

### Craft CMS version

5.3.1

### Plugin version

dev-craft-5 as 3.0.0

### Multi-site?

No

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/snipcart:"dev-craft-5 as 3.0.0"`.

**Comment by engram-design** [comment]
Fixed in [3.0.1](https://github.com/verbb/snipcart/releases/tag/3.0.1)
